docker-compose.yml

```diff
@@ -0,0 +1,155 @@
+version: '3.1'
+services:
+  mongodb:
+    image: mongo:3.4
+    restart: "no"
+    command: mongod --smallfiles
+    container_name: mongodb
+    volumes:
+      - ./mongo-data:/data/db
+      - ./mongo-app:/var/www/html
+    env_file:
+      - ./.env
+    ports:
+      - 27017:27017
+    networks:
+      mongo_net:
+
+  mongo-express:
+    image: mongo-express
+    container_name: mongoexpressweb
+    restart: "no"
+    depends_on:
+      - mongodb
+    ports:
+      - 8081:8081
+    env_file:
+      - ./.env
+    networks:
+      mongo_net:
+
+  web:
+    image: app
+    restart:  unless-stopped
+    container_name: djangoapp
+    command: python manage.py runserver 0.0.0.0:8000
+    volumes:
+      - .:/mentisparchment_docker
+    ports:
+      - 8000:8000
+    links:
+      - mongodb
+    depends_on:
+      - migration
+    env_file:
+      - ./.env
+    networks:
+      mongo_net:
+
+  migration:
+    build: .
+    image: app
+    command: python manage.py migrate
+    volumes:
+      - .:/mentisparchment_docker
+    links:
+      - mongodb
+    depends_on:
+      - make_migrations
+    env_file:
+      - ./.env
+    networks:
+      mongo_net:
+
+  make_migrations:
+    build: .
+    image: app
+    command: python manage.py makemigrations
+    volumes:
+      - .:/mentisparchment_docker
+    links:
+      - mongodb
+    depends_on:
+      - mongodb
+    env_file:
+      - ./.env
+    networks:
+      mongo_net:
+
+  es01:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.1
+    container_name: es01
+    environment:
+      - node.name=es01
+      - cluster.name=es-docker-cluster
+#      - discovery.seed_hosts=es02,es03
+      - cluster.initial_master_nodes=es01
+#      - cluster.initial_master_nodes=es01,es02,es03
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    volumes:
+      - data01:/usr/share/elasticsearch/data
+    ports:
+      - 9200:9200
+    networks:
+      - mongo_net
+    depends_on:
+      - mongodb
+    restart:  unless-stopped
+#    es02:
```

Comment:
Will we need these commented out lines later on? They seem to be exact replications of the "es01" one.

Reply:
es02 and es03 are another nodes of elasticsearch. If we will decide to enable cluster mode instead of using single node(current mode), we only will need to enable es02 and es03.

Reply:
Yeah i understand but since these two es02 and es03 blocks looked like exact copies of the other es01 block, i thought it wouldn't be hard to rewrite them anyway. I just have a thing against commented out code i guess. 😛 Not important really.